### ISSUE TITLE
Bug: GLProfiler found error 1282 - invalid operation in BDXDefaultShader.render

### ISSUE THREAD

**Issue by rafcolson** [GoranM/bdx#498]
I found a bug in [```BDXShaderProvider.java```](https://github.com/GoranM/bdx/blob/master/src/com/nilunder/bdx/gl/BDXShaderProvider.java#L46-L52) using [GLProfiler](https://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/graphics/profiling/GLProfiler.html) (only with latest LibGdx update).

Replacing ```set(u_tintColor, 0, 0, 0);``` by ```set(u_tintColor, new Color());``` seemed to have fixed it. I also added a couple of ```if```'s to prevent that attributes are set twice.

I've added this issue (instead of only submitting the PR) to introduce the GLProfiler. In BDX GL related errors are not (always?) reported. Fortunately with the new updated LibGdx, the GLProfiler, when enabled, reports such errors.

For more info, see: [LibGdx - Profiling: Error checking (since 1.6.5)](https://github.com/libgdx/libgdx/wiki/Profiling).

Edit: If you want to look for yourself, just replace ```Sacky.java``` with:

```java
import com.badlogic.gdx.graphics.profiling.GLProfiler;
import com.badlogic.gdx.graphics.profiling.GLErrorListener;

import com.nilunder.bdx.*;

public class Sacky extends GameObject{

	public void init(){
		GLProfiler.listener = GLErrorListener.THROWING_LISTENER;
		GLProfiler.enable();
    }
	
}
```

**Comment by GoranM** [comment]
Good to know. Thanks.

**Comment by SolarLune** [comment]
Ah, thanks!

It might be a good idea to add the GLProfiler to our Profiler class and enable it when the profiler is visible. It'd allow us to add in those nice pieces of information from GLProfiler (amount of draw calls, number of shader switches, vertex count, triangle count, etc).

**Comment by rafcolson** [comment]
Yes that might be a good idea. However I feel we should keep gl related profiling separate or as an additional feature. Also, I'd like to separate the code for calculating the frame rate. I think it would be useful to be able to just display the average frame rate and also to get it through a method, without having to turn on the profiler. I'll let you know if I got some ideas, in an other issue. Closing this one.
